ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/BlockSimulationResult.java

```diff
@@ -44,15 +41,36 @@ public BlockBody getBlockBody() {
     return block.getBody();
   }
 
-  public List<? extends TransactionReceipt> getReceipts() {
-    return receipts;
+  public List<TransactionReceipt> getReceipts() {
+    return blockCallSimulationResult.getReceipts();
   }
 
   public List<TransactionSimulatorResult> getTransactionSimulations() {
-    return transactionSimulationResults;
+    return blockCallSimulationResult.getTransactionSimulationResults();
   }
 
   public Block getBlock() {
     return block;
   }
+
+  public List<LogWithMetadata> getLogsWithMetadata() {
+    List<LogWithMetadata> logs = new ArrayList<>();
+    blockCallSimulationResult
+        .getTransactionSimulatorResults()
+        .forEach(
+            transactionSimulation ->
+                logs.addAll(
+                    LogWithMetadata.generate(
+                        0,
+                        transactionSimulation.logs(),
+                        block.getHeader().getNumber(),
+                        block.getHash(),
+                        transactionSimulation.result().transaction().getHash(),
+                        block
+                            .getBody()
+                            .getTransactions()
+                            .indexOf(transactionSimulation.result().transaction()),
+                        false)));
+    return logs;
+  }
```

Comment:
nit: use `.map(...).toList()` instead of creating the list outside the stream

Reply:
done